YoutubeDownloader/Views/Dialogs/AuthSetupView.xaml.cs

```diff
@@ -1,26 +1,33 @@
 ﻿using System;
 using System.Linq;
 using System.Net;
-using System.Windows;
+using Avalonia.Interactivity;
 using Microsoft.Web.WebView2.Core;
+using WebViewCore.Events;
 using YoutubeDownloader.ViewModels.Dialogs;
+using YoutubeDownloader.Views.Framework;
 
 namespace YoutubeDownloader.Views.Dialogs;
 
-public partial class AuthSetupView
+public partial class AuthSetupView : ViewModelAwareUserControl
 {
     private const string HomePageUrl = "https://www.youtube.com";
     private static readonly string LoginPageUrl =
         $"https://accounts.google.com/ServiceLogin?continue={WebUtility.UrlEncode(HomePageUrl)}";
 
-    private AuthSetupViewModel ViewModel => (AuthSetupViewModel)DataContext;
+    private AuthSetupViewModel ViewModel => (AuthSetupViewModel)DataContext!;
```

Comment:
Do you think `ViewModelAwareUserControl` could be made generic to avoid the need for this property?

Reply:
done